diff -Nru groff-1.16.1/Makefile.comm groff-1.16.1.new/Makefile.comm
--- groff-1.16.1/Makefile.comm	Tue Dec  5 09:02:32 2000
+++ groff-1.16.1.new/Makefile.comm	Tue Dec  5 08:33:29 2000
@@ -52,6 +52,10 @@
 all install install_bin install_data TAGS depend distfiles uninstall_sub:
 
 install: prefix_must_exist install_bin install_data
+	make -C $(top_srcdir)/doc install \
+		DESTDIR="$(DESTDIR)" \
+		infodir="$(infodir)"
+
 uninstall: uninstall_sub
 pure:
 
@@ -144,29 +148,25 @@
 
 .PHONY: install_man
 install_man:
-	-test -d $(manroot) || $(mkinstalldirs) $(manroot)
-	-test -d $(man1dir) || $(mkinstalldirs) $(man1dir)
+	$(mkinstalldirs) $(DESTDIR)$(man1dir)
+	$(mkinstalldirs) $(DESTDIR)$(man5dir)
+	$(mkinstalldirs) $(DESTDIR)$(man7dir)
 	@-pages="$(MAN1)"; \
 	for p in $$pages; do \
 	 prog=`basename $$p .n`; \
-	 target=$(man1dir)/$(NAMEPREFIX)$$prog.$(man1ext); \
-	 rm -f $$target; \
+	 target=$(DESTDIR)$(man1dir)/$(NAMEPREFIX)$$prog.$(man1ext); \
 	 echo $(INSTALL_DATA) $$p $$target; \
 	 $(INSTALL_DATA) $$p $$target; \
 	done
-	-test -d $(man5dir) || $(mkinstalldirs) $(man5dir)
 	@-pages="$(MAN5)"; \
 	for p in $$pages; do \
-	 target=$(man5dir)/`basename $$p .n`.$(man5ext); \
-	 rm -f $$target; \
+	 target=$(DESTDIR)$(man5dir)/`basename $$p .n`.$(man5ext); \
 	 echo $(INSTALL_DATA) $$p $$target; \
 	 $(INSTALL_DATA) $$p $$target; \
 	done
-	-test -d $(man7dir) || $(mkinstalldirs) $(man7dir)
 	@-pages="$(MAN7)"; \
 	for p in $$pages; do \
-	 target=$(man7dir)/`basename $$p .n`.$(man7ext); \
-	 rm -f $$target; \
+	 target=$(DESTDIR)$(man7dir)/`basename $$p .n`.$(man7ext); \
 	 echo $(INSTALL_DATA) $$p $$target; \
 	 $(INSTALL_DATA) $$p $$target; \
 	done
@@ -175,59 +175,54 @@
 uninstall_man:
 	@-pages="$(MAN1)"; \
 	for p in $$pages; do \
-	 target=$(man1dir)/$(NAMEPREFIX)`basename $$p .n`.$(man1ext); \
+	 target=$(DESTDIR)$(man1dir)/$(NAMEPREFIX)`basename $$p .n`.$(man1ext); \
 	 echo rm -f $$target; \
 	 rm -f $$target; \
 	done
 	@-pages="$(MAN5)"; \
 	for p in $$pages; do \
-	 target=$(man5dir)/`basename $$p .n`.$(man5ext); \
+	 target=$(DESTDIR)$(man5dir)/`basename $$p .n`.$(man5ext); \
 	 echo rm -f $$target; \
 	 rm -f $$target; \
 	done
 	@-pages="$(MAN7)"; \
 	for p in $$pages; do \
-	 target=$(man7dir)/`basename $$p .n`.$(man7ext); \
+	 target=$(DESTDIR)$(man7dir)/`basename $$p .n`.$(man7ext); \
 	 echo rm -f $$target; \
 	 rm -f $$target; \
 	done
 
 .PHONY: install_prog
 install_prog:
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	rm -f $(bindir)/$(NAMEPREFIX)$(PROG)
-	$(INSTALL_PROGRAM) $(PROG) $(bindir)/$(NAMEPREFIX)$(PROG)
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(INSTALL_PROGRAM) $(PROG) $(DESTDIR)$(bindir)/$(NAMEPREFIX)$(PROG)
 
 .PHONY: uninstall_prog
 uninstall_prog:
-	-rm -f $(bindir)/$(NAMEPREFIX)$(PROG)	
+	-rm -f $(DESTDIR)$(bindir)/$(NAMEPREFIX)$(PROG)
 
 .PHONY: install_dev
 install_dev:
-	-test -d $(datadir) || $(mkinstalldirs) $(datadir)
-	-test -d $(datasubdir) || $(mkinstalldirs) $(datasubdir)
-	-test -d $(fontdir) || $(mkinstalldirs) $(fontdir)
-	-test -d $(fontsubdir) || $(mkinstalldirs) $(fontsubdir)
-	-if test -d $(srcdir)/generate; then \
-	  test -d $(fontsubdir)/generate || \
-	    $(mkinstalldirs) $(fontsubdir)/generate; \
-	fi
-	-for f in $(DEVFILES); do \
-	  rm -f $(fontsubdir)/$$f; \
+	$(mkinstalldirs) $(DESTDIR)$(datadir)
+	$(mkinstalldirs) $(DESTDIR)$(datasubdir)
+	$(mkinstalldirs) $(DESTDIR)$(fontdir)
+	$(mkinstalldirs) $(DESTDIR)$(fontsubdir)/generate
+	for f in $(DEVFILES); do \
+	  rm -f $(DESTDIR)$(fontsubdir)/$$f; \
 	  if test -f $$f; then \
-	    $(INSTALL_DATA) $$f $(fontsubdir)/$$f; \
+	    $(INSTALL_DATA) $$f $(DESTDIR)$(fontsubdir)/$$f; \
 	  else \
-	    $(INSTALL_DATA) $(srcdir)/$$f $(fontsubdir)/$$f; \
+	    $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)$(fontsubdir)/$$f; \
 	  fi; \
 	done
 
 .PHONY: uninstall_dev
 uninstall_dev:
-	-for f in $(DEVFILES); do rm -f $(fontsubdir)/$$f; done
+	-for f in $(DEVFILES); do rm -f $(DESTDIR)$(fontsubdir)/$$f; done
 	-if test -d $(fontsubdir)/generate; then \
-	  rmdir $(fontsubdir)/generate; \
+	  rmdir $(DESTDIR)$(fontsubdir)/generate; \
 	fi
-	-rmdir $(fontsubdir)
+	-rmdir $(DESTDIR)$(fontsubdir)
 
 .PHONY: depend_src
 depend_src: depend.temp
diff -Nru groff-1.16.1/Makefile.in groff-1.16.1.new/Makefile.in
--- groff-1.16.1/Makefile.in	Tue Dec  5 09:02:32 2000
+++ groff-1.16.1.new/Makefile.in	Tue Dec  5 09:02:19 2000
@@ -149,6 +149,8 @@
 man7ext=7
 man7dir=$(manroot)/man$(man7ext)
 
+infodir=@infodir@
+
 # DEFINES should include the following:
 # -DHAVE_MMAP			if you have mmap() and <sys/mman.h>
 # -DARRAY_DELETE_NEEDS_SIZE	if your C++ doesn't understand `delete []'
@@ -272,6 +274,7 @@
   "man5dir=$(man5dir)" \
   "man7ext=$(man7ext)" \
   "man7dir=$(man7dir)" \
+  "infodir=$(infodir)" \
   "mkinstalldirs=$(mkinstalldirs)" \
   "tmac_wrap=$(tmac_wrap)" \
   "sys_tmac_prefix=$(sys_tmac_prefix)" \
diff -Nru groff-1.16.1/contrib/mm/Makefile.sub groff-1.16.1.new/contrib/mm/Makefile.sub
--- groff-1.16.1/contrib/mm/Makefile.sub	Tue Dec  5 09:02:32 2000
+++ groff-1.16.1.new/contrib/mm/Makefile.sub	Tue Dec  5 08:33:29 2000
@@ -13,25 +13,22 @@
 install: install_mm
 
 install_mm: install_mmroff install_m
-	-test -d $(tmacdir)/mm || $(mkinstalldirs) $(tmacdir)/mm
+	$(mkinstalldirs) $(DESTDIR)$(tmacdir)/mm
 	-for f in $(FILES); do \
-		rm -f $(tmacdir)/mm/$$f; \
-		$(INSTALL_DATA) $(srcdir)/mm/$$f $(tmacdir)/mm/$$f; \
+		$(INSTALL_DATA) $(srcdir)/mm/$$f $(DESTDIR)$(tmacdir)/mm/$$f; \
 	done
 	-for f in $(LOCALE); do \
-		test -f $(tmacdir)/mm/$$f || touch $(tmacdir)/mm/$$f; \
+		test -f $(tmacdir)/mm/$$f || touch $(DESTDIR)$(tmacdir)/mm/$$f; \
 	done
 
 install_m:
-	-test -d $(tmacdir) || $(mkinstalldirs) $(tmacdir)
-	-rm -f $(tmacdir)/tmac.$(tmac_m_prefix)m
-	$(INSTALL_DATA) $(srcdir)/tmac.m $(tmacdir)/tmac.$(tmac_m_prefix)m
+	$(mkinstalldirs) $(DESTDIR)$(tmacdir)
+	$(INSTALL_DATA) $(srcdir)/tmac.m $(DESTDIR)$(tmacdir)/tmac.$(tmac_m_prefix)m
 	@sed -e "s;^.mso tmac.m;.mso tmac.$(tmac_m_prefix)m;g" \
-		$(srcdir)/tmac.mse > $(tmacdir)/tmac.$(tmac_m_prefix)mse
+		$(srcdir)/tmac.mse > $(DESTDIR)$(tmacdir)/tmac.$(tmac_m_prefix)mse
 install_mmroff : mmroff
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	-rm -f $(bindir)/mmroff
-	$(INSTALL_SCRIPT) mmroff $(bindir)/mmroff
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(INSTALL_SCRIPT) mmroff $(DESTDIR)$(bindir)/mmroff
 
 mmroff: mmroff.pl
 	-rm -f $@
@@ -40,11 +37,11 @@
 
 
 uninstall_sub:
-	-for f in $(FILES); do rm -f $(tmacdir)/mm/$$f; done
+	-for f in $(FILES); do rm -f $(DESTDIR)$(tmacdir)/mm/$$f; done
 	-for f in $(LOCALE); do \
-	test -s $(tmacdir)/mm/$$f || rm -f $(tmacdir)/mm/$$f; \
+	test -s $(DESTDIR)$(tmacdir)/mm/$$f || rm -f $(DESTDIR)$(tmacdir)/mm/$$f; \
 	done
-	-rm -f $(tmacdir)/tmac.$(tmac_m_prefix)m
-	-rm -f $(tmacdir)/tmac.$(tmac_m_prefix)mse
-	-rmdir $(tmacdir)/mm
-	-rm -f $(bindir)/mmroff
+	-rm -f $(DESTDIR)$(tmacdir)/tmac.$(tmac_m_prefix)m
+	-rm -f $(DESTDIR)$(tmacdir)/tmac.$(tmac_m_prefix)mse
+	-rmdir $(DESTDIR)$(tmacdir)/mm
+	-rm -f $(DESTDIR)$(bindir)/mmroff
diff -Nru groff-1.16.1/doc/Makefile groff-1.16.1.new/doc/Makefile
--- groff-1.16.1/doc/Makefile	Tue Dec  5 09:02:32 2000
+++ groff-1.16.1.new/doc/Makefile	Tue Dec  5 08:33:29 2000
@@ -68,7 +68,9 @@
 groff: groff.texinfo
 	makeinfo groff.texinfo
 
-install:
+install: groff
+	$(mkinstalldirs) $(DESTDIR)$(infodir)
+	$(INSTALL_DATA) groff.info* $(DESTDIR)$(infodir)
 
 clean:
 	-rm -f *.ps *.html *.ascii *.png *.gif *.dit core
diff -Nru groff-1.16.1/src/preproc/eqn/Makefile.sub groff-1.16.1.new/src/preproc/eqn/Makefile.sub
--- groff-1.16.1/src/preproc/eqn/Makefile.sub	Tue Dec  5 09:02:32 2000
+++ groff-1.16.1.new/src/preproc/eqn/Makefile.sub	Tue Dec  5 08:33:29 2000
@@ -50,8 +50,7 @@
 	chmod +x neqn
 
 install_data: neqn
-	-rm -f $(bindir)/$(NAMEPREFIX)neqn
-	$(INSTALL_SCRIPT) neqn $(bindir)/$(NAMEPREFIX)neqn
+	$(INSTALL_SCRIPT) neqn $(DESTDIR)$(bindir)/$(NAMEPREFIX)neqn
 
 uninstall_sub:
-	-rm -f $(bindir)/$(NAMEPREFIX)neqn
+	-rm -f $(DESTDIR)$(bindir)/$(NAMEPREFIX)neqn
diff -Nru groff-1.16.1/src/roff/grog/Makefile.sub groff-1.16.1.new/src/roff/grog/Makefile.sub
--- groff-1.16.1/src/roff/grog/Makefile.sub	Tue Dec  5 09:02:32 2000
+++ groff-1.16.1.new/src/roff/grog/Makefile.sub	Tue Dec  5 08:33:29 2000
@@ -16,9 +16,8 @@
 	chmod +x $@
 
 install_data: grog
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	-rm -f $(bindir)/grog
-	$(INSTALL_SCRIPT) grog $(bindir)/grog
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(INSTALL_SCRIPT) grog $(DESTDIR)$(bindir)/grog
 
 uninstall_sub:
-	-rm -f $(bindir)/grog
+	-rm -f $(DESTDIR)$(bindir)/grog
diff -Nru groff-1.16.1/src/roff/nroff/Makefile.sub groff-1.16.1.new/src/roff/nroff/Makefile.sub
--- groff-1.16.1/src/roff/nroff/Makefile.sub	Tue Dec  5 09:02:32 2000
+++ groff-1.16.1.new/src/roff/nroff/Makefile.sub	Tue Dec  5 08:33:29 2000
@@ -10,9 +10,8 @@
 	chmod +x $@
 
 install_data: nroff
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	-rm -f $(bindir)/$(NAMEPREFIX)nroff
-	$(INSTALL_SCRIPT) nroff $(bindir)/$(NAMEPREFIX)nroff
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(INSTALL_SCRIPT) nroff $(DESTDIR)$(bindir)/$(NAMEPREFIX)nroff
 
 uninstall_sub:
-	-rm -f $(bindir)/$(NAMEPREFIX)nroff
+	-rm -f $(DESTDIR)$(bindir)/$(NAMEPREFIX)nroff
diff -Nru groff-1.16.1/src/roff/troff/Makefile.sub groff-1.16.1.new/src/roff/troff/Makefile.sub
--- groff-1.16.1/src/roff/troff/Makefile.sub	Tue Dec  5 09:02:32 2000
+++ groff-1.16.1.new/src/roff/troff/Makefile.sub	Tue Dec  5 08:33:29 2000
@@ -47,11 +47,8 @@
 	@echo const char \*revision = \"`cat $(top_srcdir)/REVISION`\"\; >>$@
 
 install_data: hyphen.us
-	-test -d $(datadir) || $(mkinstalldirs) $(datadir)
-	-test -d $(datasubdir) || $(mkinstalldirs) $(datasubdir)
-	-test -d $(tmacdir) || $(mkinstalldirs) $(tmacdir)
-	-rm -f $(tmacdir)/hyphen.us
-	$(INSTALL_DATA) $(srcdir)/hyphen.us $(tmacdir)/hyphen.us
+	$(mkinstalldirs) $(DESTDIR)$(tmacdir)
+	$(INSTALL_DATA) $(srcdir)/hyphen.us $(DESTDIR)$(tmacdir)/hyphen.us
 
 uninstall_sub:
-	-rm -f $(tmacdir)/hyphen.us
+	-rm -f $(DESTDIR)$(tmacdir)/hyphen.us
diff -Nru groff-1.16.1/src/utils/afmtodit/Makefile.sub groff-1.16.1.new/src/utils/afmtodit/Makefile.sub
--- groff-1.16.1/src/utils/afmtodit/Makefile.sub	Tue Dec  5 09:02:32 2000
+++ groff-1.16.1.new/src/utils/afmtodit/Makefile.sub	Tue Dec  5 08:33:29 2000
@@ -13,9 +13,8 @@
 	chmod +x afmtodit
 
 install_data: afmtodit
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	-rm -f $(bindir)/afmtodit
-	$(INSTALL_SCRIPT) afmtodit $(bindir)/afmtodit
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(INSTALL_SCRIPT) afmtodit $(DESTDIR)$(bindir)/afmtodit
 
 uninstall_sub:
-	-rm -f $(bindir)/afmtodit
+	-rm -f $(DESTDIR)$(bindir)/afmtodit
diff -Nru groff-1.16.1/src/utils/indxbib/Makefile.sub groff-1.16.1.new/src/utils/indxbib/Makefile.sub
--- groff-1.16.1/src/utils/indxbib/Makefile.sub	Tue Dec  5 09:02:32 2000
+++ groff-1.16.1.new/src/utils/indxbib/Makefile.sub	Tue Dec  5 08:33:29 2000
@@ -14,17 +14,9 @@
 NAMEPREFIX=$(g)
 
 install_data: eign
-	-test -d $(datadir) || $(mkinstalldirs) $(datadir)
-	-test -d $(datasubdir) || $(mkinstalldirs) $(datasubdir)
-	if test -f /usr/lib/eign; then \
-	  rm -f $(common_words_file); \
-	  ln -s /usr/lib/eign $(common_words_file) 2>/dev/null \
-	  || ln /usr/lib/eign $(common_words_file) 2>/dev/null \
-	  || cp /usr/lib/eign $(common_words_file); \
-	else \
-	  rm -f $(common_words_file); \
-	  $(INSTALL_DATA) $(srcdir)/eign $(common_words_file); \
-	fi
+	$(mkinstalldirs) $(DESTDIR)$(datadir)
+	$(mkinstalldirs) $(DESTDIR)$(datasubdir)
+	$(INSTALL_DATA) $(srcdir)/eign $(DESTDIR)$(common_words_file)
 
 uninstall_sub:
-	-rm -f $(common_words_file)
+	-rm -f $(DESTDIR)$(common_words_file)
diff -Nru groff-1.16.1/tmac/Makefile.sub groff-1.16.1.new/tmac/Makefile.sub
--- groff-1.16.1/tmac/Makefile.sub	Tue Dec  5 09:02:32 2000
+++ groff-1.16.1.new/tmac/Makefile.sub	Tue Dec  5 08:33:29 2000
@@ -31,36 +31,25 @@
 
 install_data: $(NORMALFILES) $(SPECIALFILES) \
               stamp-wrap stamp-strip stamp-sed man.local
-	-test -d $(tmacdir) || $(mkinstalldirs) $(tmacdir)
-	if test -n "$(tmac_wrap)"; then \
-	  for m in ""$(tmac_wrap); do \
-	    $(INSTALL_DATA) $$m-wrap $(tmacdir)/tmac.$$m; \
-	  done; \
-	fi
+	$(mkinstalldirs) $(DESTDIR)$(tmacdir)
+	$(mkinstalldirs) $(DESTDIR)$(mdocdir)
+	for m in ""$(tmac_wrap); do \
+	    $(INSTALL_DATA) $$m-wrap $(DESTDIR)$(tmacdir)/tmac.$$m; \
+	done; \
 	for f in $(NORMALFILES); do \
-	  rm -f $(tmacdir)/$$f; \
-	  $(INSTALL_DATA) $(srcdir)/$$f $(tmacdir)/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)$(tmacdir)/$$f; \
 	done
-	-rm -f $(tmacdir)/tmac.$(tmac_s_prefix)s
-	$(INSTALL_DATA) $(srcdir)/tmac.s $(tmacdir)/tmac.$(tmac_s_prefix)s
-	-rm -f $(tmacdir)/tmac.ms
-	$(INSTALL_DATA) tmac.ms-sed $(tmacdir)/tmac.ms
-	-rm -f $(tmacdir)/tmac.$(tmac_an_prefix)an
-	$(INSTALL_DATA) $(srcdir)/tmac.an $(tmacdir)/tmac.$(tmac_an_prefix)an
-	-rm -f $(tmacdir)/tmac.man
-	$(INSTALL_DATA) tmac.man-sed $(tmacdir)/tmac.man
+	$(INSTALL_DATA) $(srcdir)/tmac.s $(DESTDIR)$(tmacdir)/tmac.$(tmac_s_prefix)s
+	$(INSTALL_DATA) tmac.ms-sed $(DESTDIR)$(tmacdir)/tmac.ms
+	$(INSTALL_DATA) $(srcdir)/tmac.an $(DESTDIR)$(tmacdir)/tmac.$(tmac_an_prefix)an
+	$(INSTALL_DATA) tmac.man-sed $(DESTDIR)$(tmacdir)/tmac.man
 	for f in $(STRIPFILES); do \
-	  rm -f $(tmacdir)/$$f; \
-	  $(INSTALL_DATA) $$f-s $(tmacdir)/$$f; \
+	  $(INSTALL_DATA) $$f-s $(DESTDIR)$(tmacdir)/$$f; \
 	done
-	-test -d $(mdocdir) || $(mkinstalldirs) $(mdocdir)
 	for f in $(MDOCFILES); do \
-	  rm -f $(mdocdir)/$$f; \
-	  $(INSTALL_DATA) $$f-s $(mdocdir)/$$f; \
+	  $(INSTALL_DATA) $$f-s $(DESTDIR)$(mdocdir)/$$f; \
 	done
-	-test -f $(tmacdir)/man.local || \
-	  $(INSTALL_DATA) $(srcdir)/man.local $(tmacdir)/man.local
-	-rm -f temp
+	$(INSTALL_DATA) $(srcdir)/man.local $(DESTDIR)$(tmacdir)/man.local
 
 stamp-strip: $(STRIPFILES) $(MDOCFILES)
 	for f in $(STRIPFILES) $(MDOCFILES); do \
@@ -89,17 +78,17 @@
 	touch $@
 
 uninstall_sub:
-	-if test -n "$(tmac_wrap)"; then \
+	-if test -n "$(DESTDIR)$(tmac_wrap)"; then \
 	  for m in ""$(tmac_wrap); do \
-	    rm -f $(tmacdir)/tmac.$$m; \
+	    rm -f $(DESTDIR)$(tmacdir)/tmac.$$m; \
 	  done; \
 	fi
-	-for f in $(NORMALFILES) $(STRIPFILES); do rm -f $(tmacdir)/$$f; done
-	-rm -f $(tmacdir)/tmac.$(tmac_s_prefix)s
-	-rm -f $(tmacdir)/tmac.$(tmac_an_prefix)an
-	-rm -f $(tmacdir)/tmac.man $(tmacdir)/tmac.ms
-	-if cmp -s $(tmacdir)/man.local $(srcdir)/man.local; then \
-	  rm -f $(tmacdir)/man.local; \
+	-for f in $(NORMALFILES) $(STRIPFILES); do rm -f $(DESTDIR)$(tmacdir)/$$f; done
+	-rm -f $(DESTDIR)$(tmacdir)/tmac.$(tmac_s_prefix)s
+	-rm -f $(DESTDIR)$(tmacdir)/tmac.$(tmac_an_prefix)an
+	-rm -f $(DESTDIR)$(tmacdir)/tmac.man $(tmacdir)/tmac.ms
+	-if cmp -s $(DESTDIR)$(tmacdir)/man.local $(srcdir)/man.local; then \
+	  rm -f $(DESTDIR)$(tmacdir)/man.local; \
 	fi
-	-for f in $(MDOCFILES); do rm -f $(mdocdir)/$$f; done
-	-rmdir $(mdocdir)
+	-for f in $(MDOCFILES); do rm -f $(DESTDIR)$(mdocdir)/$$f; done
+	-rmdir $(DESTDIR)$(mdocdir)
