--- groff-1.19.1/Makefile.comm.orig	2004-03-09 07:47:58.000000000 +0100
+++ groff-1.19.1/Makefile.comm	2004-07-20 08:08:52.291966144 +0200
@@ -168,29 +168,25 @@
 
 .PHONY: install_man
 install_man:
-	-test -d $(manroot) || $(mkinstalldirs) $(manroot)
-	-test -d $(man1dir) || $(mkinstalldirs) $(man1dir)
+	$(mkinstalldirs) $(DESTDIR)$(man1dir)
+	$(mkinstalldirs) $(DESTDIR)$(man5dir)
+	$(mkinstalldirs) $(DESTDIR)$(man7dir)
 	@-pages="$(MAN1)"; \
 	for p in $$pages; do \
 	 prog=`basename $$p .n`; \
-	 target=$(man1dir)/$(NAMEPREFIX)$$prog.$(man1ext); \
-	 rm -f $$target; \
+	 target=$(DESTDIR)$(man1dir)/$(NAMEPREFIX)$$prog.$(man1ext); \
 	 echo $(INSTALL_DATA) $$p $$target; \
 	 $(INSTALL_DATA) $$p $$target; \
 	done
-	-test -d $(man5dir) || $(mkinstalldirs) $(man5dir)
 	@-pages="$(MAN5)"; \
 	for p in $$pages; do \
-	 target=$(man5dir)/`basename $$p .n`.$(man5ext); \
-	 rm -f $$target; \
+	 target=$(DESTDIR)$(man5dir)/`basename $$p .n`.$(man5ext); \
 	 echo $(INSTALL_DATA) $$p $$target; \
 	 $(INSTALL_DATA) $$p $$target; \
 	done
-	-test -d $(man7dir) || $(mkinstalldirs) $(man7dir)
 	@-pages="$(MAN7)"; \
 	for p in $$pages; do \
-	 target=$(man7dir)/`basename $$p .n`.$(man7ext); \
-	 rm -f $$target; \
+	 target=$(DESTDIR)$(man7dir)/`basename $$p .n`.$(man7ext); \
 	 echo $(INSTALL_DATA) $$p $$target; \
 	 $(INSTALL_DATA) $$p $$target; \
 	done
@@ -218,9 +214,8 @@
 
 .PHONY: install_prog
 install_prog:
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	rm -f $(bindir)/$(NAMEPREFIX)$(PROG)
-	$(INSTALL_PROGRAM) $(PROG) $(bindir)/$(NAMEPREFIX)$(PROG)
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(INSTALL_PROGRAM) $(PROG) $(DESTDIR)$(bindir)/$(NAMEPREFIX)$(PROG)
 
 .PHONY: uninstall_prog
 uninstall_prog:
@@ -228,31 +223,31 @@
 
 .PHONY: install_dev
 install_dev:
-	-test -d $(datadir) || $(mkinstalldirs) $(datadir)
-	-test -d $(dataprogramdir) || $(mkinstalldirs) $(dataprogramdir)
-	-test -d $(datasubdir) || $(mkinstalldirs) $(datasubdir)
-	-test -d $(fontdir) || $(mkinstalldirs) $(fontdir)
-	-test -d $(fontsubdir) || $(mkinstalldirs) $(fontsubdir)
+	-test -d $(DESTDIR)$(datadir) || $(mkinstalldirs) $(DESTDIR)$(datadir)
+	-test -d $(DESTDIR)$(dataprogramdir) || $(mkinstalldirs) $(DESTDIR)$(dataprogramdir)
+	-test -d $(DESTDIR)$(datasubdir) || $(mkinstalldirs) $(DESTDIR)$(datasubdir)
+	-test -d $(DESTDIR)$(fontdir) || $(mkinstalldirs) $(DESTDIR)$(fontdir)
+	-test -d $(DESTDIR)$(fontsubdir) || $(mkinstalldirs) $(DESTDIR)$(fontsubdir)
 	-if test -d $(srcdir)/generate; then \
-	  test -d $(fontsubdir)/generate \
-	    || $(mkinstalldirs) $(fontsubdir)/generate; \
+	  test -d $(DESTDIR)$(fontsubdir)/generate \
+	    || $(mkinstalldirs) $(DESTDIR)$(fontsubdir)/generate; \
 	fi
 	-test -z "$(DEVFILES)" \
 	  || for f in ""$(DEVFILES); do \
-	       rm -f $(fontsubdir)/$$f; \
+	       rm -f $(DESTDIR)$(fontsubdir)/$$f; \
 	       if test -f $$f; then \
-		 $(INSTALL_DATA) $$f $(fontsubdir)/$$f; \
+		 $(INSTALL_DATA) $$f $(DESTDIR)$(fontsubdir)/$$f; \
 	       else \
-		 $(INSTALL_DATA) $(srcdir)/$$f $(fontsubdir)/$$f; \
+		 $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)$(fontsubdir)/$$f; \
 	       fi; \
 	     done
 	-test -z "$(DEVSCRIPTS)" \
 	  || for f in ""$(DEVSCRIPTS); do \
-	       rm -f $(fontsubdir)/$$f; \
+	       rm -f $(DESTDIR)$(fontsubdir)/$$f; \
 	       if test -f $$f; then \
-		 $(INSTALL_SCRIPT) $$f $(fontsubdir)/$$f; \
+		 $(INSTALL_SCRIPT) $$f $(DESTDIR)$(fontsubdir)/$$f; \
 	       else \
-		 $(INSTALL_SCRIPT) $(srcdir)/$$f $(fontsubdir)/$$f; \
+		 $(INSTALL_SCRIPT) $(srcdir)/$$f $(DESTDIR)$(fontsubdir)/$$f; \
 	       fi; \
 	     done
 
diff -Nru groff-1.19.orig/contrib/eqn2graph/Makefile.sub groff-1.19/contrib/eqn2graph/Makefile.sub
--- groff-1.19.orig/contrib/eqn2graph/Makefile.sub	2002-07-08 00:14:56.000000000 +0200
+++ groff-1.19/contrib/eqn2graph/Makefile.sub	2003-04-30 09:40:08.000000000 +0200
@@ -11,9 +11,8 @@
 	chmod +x $@
 
 install_data: eqn2graph
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	-rm -f $(bindir)/eqn2graph
-	$(INSTALL_SCRIPT) eqn2graph $(bindir)/eqn2graph
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(INSTALL_SCRIPT) eqn2graph $(DESTDIR)$(bindir)/eqn2graph
 
 uninstall_sub:
 	-rm -f $(bindir)/eqn2graph
--- groff-1.19.2/contrib/groffer/Makefile.sub.orig	2005-08-15 16:49:38.000000000 +0200
+++ groff-1.19.2/contrib/groffer/Makefile.sub	2007-06-18 21:15:28.894390694 +0200
@@ -45,14 +45,13 @@
 	chmod +x $@
 
 install_data: groffer
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	-$(RM) $(bindir)/groffer
-	$(INSTALL_SCRIPT) groffer $(bindir)/groffer
-	-test -d $(libdir)/groff/groffer || \
-          $(mkinstalldirs) $(libdir)/groff/groffer
-	-$(RM) $(libdir)/groff/groffer/groffer2.sh
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	-$(RM) $(DESTDIR)$(bindir)/groffer
+	$(INSTALL_SCRIPT) groffer $(DESTDIR)$(bindir)/groffer
+	$(mkinstalldirs) $(DESTDIR)$(libdir)/groff/groffer
+	-$(RM) $(DESTDIR)$(libdir)/groff/groffer/groffer2.sh
 	$(INSTALL_SCRIPT) $(srcdir)/groffer2.sh \
-	  $(libdir)/groff/groffer/groffer2.sh
+	  $(DESTDIR)$(libdir)/groff/groffer/groffer2.sh
 
 uninstall_sub:
 	-$(RM) $(bindir)/groffer
diff -Nru groff-1.19.orig/contrib/mm/Makefile.sub groff-1.19/contrib/mm/Makefile.sub
--- groff-1.19.orig/contrib/mm/Makefile.sub	2002-05-22 08:26:12.000000000 +0200
+++ groff-1.19/contrib/mm/Makefile.sub	2003-04-30 09:40:08.000000000 +0200
@@ -17,31 +17,27 @@
 install: install_mm
 
 install_mm: install_mmroff install_m
-	-test -d $(tmacdir)/mm || $(mkinstalldirs) $(tmacdir)/mm
+	$(mkinstalldirs) $(DESTDIR)$(tmacdir)/mm
 	-for f in $(FILES); do \
-		rm -f $(tmacdir)/mm/$$f; \
-		$(INSTALL_DATA) $(srcdir)/mm/$$f $(tmacdir)/mm/$$f; \
+		$(INSTALL_DATA) $(srcdir)/mm/$$f $(DESTDIR)$(tmacdir)/mm/$$f; \
 	done
 	-for f in $(LOCALE); do \
-		test -f $(tmacdir)/mm/$$f || touch $(tmacdir)/mm/$$f; \
+		test -f $(DESTDIR)$(tmacdir)/mm/$$f || touch $(DESTDIR)$(tmacdir)/mm/$$f; \
 	done
 
 install_m:
-	-test -d $(tmacdir) || $(mkinstalldirs) $(tmacdir)
-	-rm -f $(tmacdir)/tmac.$(tmac_m_prefix)m
-	-rm -f $(tmacdir)/$(tmac_m_prefix)m.tmac
-	$(INSTALL_DATA) $(srcdir)/m.tmac $(tmacdir)/$(tmac_m_prefix)m.tmac
+	$(mkinstalldirs) $(DESTDIR)$(tmacdir)
+	$(INSTALL_DATA) $(srcdir)/m.tmac $(DESTDIR)$(tmacdir)/$(tmac_m_prefix)m.tmac
 	@sed -e "s;^.mso m.tmac;.mso $(tmac_m_prefix)m.tmac;g" \
-		$(srcdir)/mse.tmac > $(tmacdir)/$(tmac_m_prefix)mse.tmac
+		$(srcdir)/mse.tmac > $(DESTDIR)$(tmacdir)/$(tmac_m_prefix)mse.tmac
 	@sed -e "s;^.mso mse.tmac;.mso $(tmac_m_prefix)mse.tmac;g" \
-		$(srcdir)/mmse.tmac > $(tmacdir)/$(tmac_m_prefix)mmse.tmac
+		$(srcdir)/mmse.tmac > $(DESTDIR)$(tmacdir)/$(tmac_m_prefix)mmse.tmac
 	@sed -e "s;^.mso m.tmac;.mso $(tmac_m_prefix)m.tmac;g" \
-		$(srcdir)/mm.tmac > $(tmacdir)/$(tmac_m_prefix)mm.tmac
+		$(srcdir)/mm.tmac > $(DESTDIR)$(tmacdir)/$(tmac_m_prefix)mm.tmac
 
 install_mmroff: mmroff
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	-rm -f $(bindir)/mmroff
-	$(INSTALL_SCRIPT) mmroff $(bindir)/mmroff
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(INSTALL_SCRIPT) mmroff $(DESTDIR)$(bindir)/mmroff
 
 mmroff: mmroff.pl
 	-rm -f $@
--- groff-1.19.2/contrib/mom/Makefile.sub.orig	2005-08-11 06:18:15.000000000 +0200
+++ groff-1.19.2/contrib/mom/Makefile.sub	2007-06-18 21:16:24.109537226 +0200
@@ -104,24 +104,24 @@
 
 install_data: $(NORMALFILES) $(HTMLDOCFILES) \
               $(EXAMPLEFILES) $(PROCESSEDEXAMPLEFILES)
-	-test -d $(tmacdir) || $(mkinstalldirs) $(tmacdir)
+	$(mkinstalldirs) $(DESTDIR)$(tmacdir)
 	for f in $(NORMALFILES); do \
-	  rm -f $(tmacdir)/$$f; \
-	  $(INSTALL_DATA) $(srcdir)/$$f $(tmacdir)/$$f; \
+	  rm -f $(DESTDIR)$(tmacdir)/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)$(tmacdir)/$$f; \
 	done
-	-test -d $(htmldocdir)/mom || $(mkinstalldirs) $(htmldocdir)/mom
+	$(mkinstalldirs) $(DESTDIR)$(htmldocdir)/mom
 	for f in $(HTMLDOCFILES_); do \
-	  rm -f $(htmldocdir)/mom/$$f; \
-	  $(INSTALL_DATA) $(srcdir)/momdoc/$$f $(htmldocdir)/mom/$$f; \
+	  rm -f $(DESTDIR)$(htmldocdir)/mom/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/momdoc/$$f $(DESTDIR)$(htmldocdir)/mom/$$f; \
 	done
-	-test -d $(exampledir)/mom || $(mkinstalldirs) $(exampledir)/mom
+	$(mkinstalldirs) $(DESTDIR)$(exampledir)/mom
 	for f in $(EXAMPLEFILES_); do \
-	  rm -f $(exampledir)/mom/$$f; \
-	  $(INSTALL_DATA) $(srcdir)/examples/$$f $(exampledir)/mom/$$f; \
+	  rm -f $(DESTDIR)$(exampledir)/mom/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/examples/$$f $(DESTDIR)$(exampledir)/mom/$$f; \
 	done
 	for f in $(PROCESSEDEXAMPLEFILES_); do \
-	  rm -f $(exampledir)/mom/$$f; \
-	  $(INSTALL_DATA) examples/$$f $(exampledir)/mom/$$f; \
+	  rm -f $(DESTDIR)$(exampledir)/mom/$$f; \
+	  $(INSTALL_DATA) examples/$$f $(DESTDIR)$(exampledir)/mom/$$f; \
 	done
 
 uninstall_sub:
diff -Nru groff-1.19.orig/contrib/pic2graph/Makefile.sub groff-1.19/contrib/pic2graph/Makefile.sub
--- groff-1.19.orig/contrib/pic2graph/Makefile.sub	2001-08-11 00:16:11.000000000 +0200
+++ groff-1.19/contrib/pic2graph/Makefile.sub	2003-04-30 09:40:08.000000000 +0200
@@ -11,9 +11,8 @@
 	chmod +x $@
 
 install_data: pic2graph
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	-rm -f $(bindir)/pic2graph
-	$(INSTALL_SCRIPT) pic2graph $(bindir)/pic2graph
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(INSTALL_SCRIPT) pic2graph $(DESTDIR)$(bindir)/pic2graph
 
 uninstall_sub:
 	-rm -f $(bindir)/pic2graph
--- groff-1.19.1/doc/Makefile.sub.orig	2004-03-05 10:52:57.000000000 +0100
+++ groff-1.19.1/doc/Makefile.sub	2004-07-20 08:11:31.467767744 +0200
@@ -150,52 +150,52 @@
 
 install_data: groff $(DOCFILES) $(PROCESSEDDOCFILES) $(make_install_html) \
               $(EXAMPLEFILES) $(PROCESSEDEXAMPLEFILES)
-	-test -d $(infodir) || $(mkinstalldirs) $(infodir)
+	-test -d $(DESTDIR)$(infodir) || $(mkinstalldirs) $(DESTDIR)$(infodir)
 # Prefer info files in builddir over srcdir; we test for
 # the existence of `groff'.
 	d=.; \
-	  test -f "groff" || d=$(srcdir); \
-	  for f in $$d/groff `ls $$d/groff-*`; do \
-	    rm -f $(infodir)/`basename $$f`; \
-	    $(INSTALL_DATA) $$f $(infodir)/`basename $$f`; \
+	  test -f "groff.info" || d=$(srcdir); \
+	  for f in $$d/groff.info `ls $$d/groff.info-*`; do \
+	    rm -f $(DESTDIR)$(infodir)/`basename $$f`; \
+	    $(INSTALL_DATA) $$f $(DESTDIR)$(infodir)/`basename $$f`; \
 	  done
-	$(INSTALL_INFO) --info-dir=$(infodir) $(infodir)/groff
-	-test -d $(docdir) || $(mkinstalldirs) $(docdir)
+	$(INSTALL_INFO) --info-dir=$(DESTDIR)$(infodir) $(DESTDIR)$(infodir)/groff.info
+	-test -d $(DESTDIR)$(docdir) || $(mkinstalldirs) $(DESTDIR)$(docdir)
 	for f in $(DOCFILES); do \
-	  rm -f $(docdir)/$$f; \
-	  $(INSTALL_DATA) $(srcdir)/$$f $(docdir)/$$f; \
+	  rm -f $(DESTDIR)$(docdir)/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)$(docdir)/$$f; \
 	done
 	for f in $(PROCESSEDDOCFILES); do \
-	  rm -f $(docdir)/$$f; \
-	  $(INSTALL_DATA) $$f $(docdir)/$$f; \
+	  rm -f $(DESTDIR)$(docdir)/$$f; \
+	  $(INSTALL_DATA) $$f $(DESTDIR)$(docdir)/$$f; \
 	done
-	-test -d $(exampledir) || $(mkinstalldirs) $(exampledir)
+	-test -d $(DESTDIR)$(exampledir) || $(mkinstalldirs) $(DESTDIR)$(exampledir)
 	for f in $(EXAMPLEFILES); do \
-	  rm -f $(exampledir)/$$f; \
-	  $(INSTALL_DATA) $(srcdir)/$$f $(exampledir)/$$f; \
+	  rm -f $(DESTDIR)$(exampledir)/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)$(exampledir)/$$f; \
 	done
 	for f in $(PROCESSEDEXAMPLEFILES); do \
-	  rm -f $(exampledir)/$$f; \
-	  $(INSTALL_DATA) $$f $(exampledir)/$$f; \
+	  rm -f $(DESTDIR)$(exampledir)/$$f; \
+	  $(INSTALL_DATA) $$f $(DESTDIR)$(exampledir)/$$f; \
 	done
 
 install_html: $(HTMLDOCFILES) $(HTMLEXAMPLEFILES)
-	-test -d $(htmldocdir) || $(mkinstalldirs) $(htmldocdir)
+	-test -d $(DESTDIR)$(htmldocdir) || $(mkinstalldirs) $(DESTDIR)$(htmldocdir)
 	for f in `ls $(HTMLDOCFILESALL)`; do \
-	  rm -f $(htmldocdir)/$$f; \
-	  $(INSTALL_DATA) $$f $(htmldocdir)/$$f; \
+	  rm -f $(DESTDIR)$(htmldocdir)/$$f; \
+	  $(INSTALL_DATA) $$f $(DESTDIR)$(htmldocdir)/$$f; \
 	done
-	-test -d $(htmldocimagedir) || $(mkinstalldirs) $(htmldocimagedir)
-	rm -f $(htmldocimagedir)/$(HTMLDOCIMAGEFILES)
-	$(INSTALL_DATA) $(imagedir)/$(HTMLDOCIMAGEFILES) $(htmldocimagedir)
-	-test -d $(exampledir) || $(mkinstalldirs) $(exampledir)
+	-test -d $(DESTDIR)$(htmldocimagedir) || $(mkinstalldirs) $(DESTDIR)$(htmldocimagedir)
+	rm -f $(DESTDIR)$(htmldocimagedir)/$(HTMLDOCIMAGEFILES)
+	$(INSTALL_DATA) $(imagedir)/$(HTMLDOCIMAGEFILES) $(DESTDIR)$(htmldocimagedir)
+	-test -d $(DESTDIR)$(exampledir) || $(mkinstalldirs) $(DESTDIR)$(exampledir)
 	for f in $(HTMLEXAMPLEFILES); do \
-	  rm -f $(exampledir)/$$f; \
-	  $(INSTALL_DATA) $$f $(exampledir)/$$f; \
+	  rm -f $(DESTDIR)$(exampledir)/$$f; \
+	  $(INSTALL_DATA) $$f $(DESTDIR)$(exampledir)/$$f; \
 	done
-	-test -d $(exampleimagedir) || $(mkinstalldirs) $(exampleimagedir)
-	rm -f $(exampleimagedir)/$(HTMLEXAMPLEIMAGEFILES)
-	$(INSTALL_DATA) $(imagedir)/$(HTMLEXAMPLEIMAGEFILES) $(exampleimagedir)
+	-test -d $(DESTDIR)$(exampleimagedir) || $(mkinstalldirs) $(DESTDIR)$(exampleimagedir)
+	rm -f $(DESTDIR)$(exampleimagedir)/$(HTMLEXAMPLEIMAGEFILES)
+	$(INSTALL_DATA) $(imagedir)/$(HTMLEXAMPLEIMAGEFILES) $(DESTDIR)$(exampleimagedir)
 
 uninstall_sub:
 	-for f in groff `ls groff-*`; do \
diff -Nru groff-1.19.orig/src/preproc/eqn/Makefile.sub groff-1.19/src/preproc/eqn/Makefile.sub
--- groff-1.19.orig/src/preproc/eqn/Makefile.sub	2003-04-12 15:52:47.000000000 +0200
+++ groff-1.19/src/preproc/eqn/Makefile.sub	2003-04-30 09:40:08.000000000 +0200
@@ -53,8 +53,7 @@
 	chmod +x $@
 
 install_data: neqn
-	-rm -f $(bindir)/$(NAMEPREFIX)neqn
-	$(INSTALL_SCRIPT) neqn $(bindir)/$(NAMEPREFIX)neqn
+	$(INSTALL_SCRIPT) neqn $(DESTDIR)$(bindir)/$(NAMEPREFIX)neqn
 
 uninstall_sub:
 	-rm -f $(bindir)/$(NAMEPREFIX)neqn
diff -Nru groff-1.19.orig/src/roff/grog/Makefile.sub groff-1.19/src/roff/grog/Makefile.sub
--- groff-1.19.orig/src/roff/grog/Makefile.sub	2002-06-14 12:44:00.000000000 +0200
+++ groff-1.19/src/roff/grog/Makefile.sub	2003-04-30 09:40:08.000000000 +0200
@@ -24,9 +24,8 @@
 	chmod +x $@
 
 install_data: grog
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	-rm -f $(bindir)/grog
-	$(INSTALL_SCRIPT) grog $(bindir)/grog
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(INSTALL_SCRIPT) grog $(DESTDIR)$(bindir)/grog
 
 uninstall_sub:
 	-rm -f $(bindir)/grog
diff -Nru groff-1.19.orig/src/roff/nroff/Makefile.sub groff-1.19/src/roff/nroff/Makefile.sub
--- groff-1.19.orig/src/roff/nroff/Makefile.sub	2003-01-28 09:40:29.000000000 +0100
+++ groff-1.19/src/roff/nroff/Makefile.sub	2003-04-30 09:40:08.000000000 +0200
@@ -13,9 +13,8 @@
 	chmod +x $@
 
 install_data: nroff
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	-rm -f $(bindir)/$(NAMEPREFIX)nroff
-	$(INSTALL_SCRIPT) nroff $(bindir)/$(NAMEPREFIX)nroff
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(INSTALL_SCRIPT) nroff $(DESTDIR)$(bindir)/$(NAMEPREFIX)nroff
 
 uninstall_sub:
 	-rm -f $(bindir)/$(NAMEPREFIX)nroff
diff -Nru groff-1.19.orig/src/utils/afmtodit/Makefile.sub groff-1.19/src/utils/afmtodit/Makefile.sub
--- groff-1.19.orig/src/utils/afmtodit/Makefile.sub	2003-02-25 17:55:24.000000000 +0100
+++ groff-1.19/src/utils/afmtodit/Makefile.sub	2003-04-30 09:40:08.000000000 +0200
@@ -17,9 +17,8 @@
 	chmod +x afmtodit
 
 install_data: afmtodit
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	-rm -f $(bindir)/afmtodit
-	$(INSTALL_SCRIPT) afmtodit $(bindir)/afmtodit
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(INSTALL_SCRIPT) afmtodit $(DESTDIR)$(bindir)/afmtodit
 
 uninstall_sub:
 	-rm -f $(bindir)/afmtodit
--- groff-1.19.2/src/utils/indxbib/Makefile.sub.orig	2004-06-03 08:24:29.000000000 +0200
+++ groff-1.19.2/src/utils/indxbib/Makefile.sub	2007-06-18 21:17:48.266333049 +0200
@@ -12,18 +12,11 @@
 NAMEPREFIX=$(g)
 
 install_data: $(srcdir)/eign
-	-test -d $(datadir) || $(mkinstalldirs) $(datadir)
-	-test -d $(dataprogramdir) || $(mkinstalldirs) $(dataprogramdir)
-	-test -d $(datasubdir) || $(mkinstalldirs) $(datasubdir)
-	if test -f /usr/lib/eign; then \
-	  rm -f $(common_words_file); \
-	  ln -s /usr/lib/eign $(common_words_file) 2>/dev/null \
-	  || ln /usr/lib/eign $(common_words_file) 2>/dev/null \
-	  || cp /usr/lib/eign $(common_words_file); \
-	else \
-	  rm -f $(common_words_file); \
-	  $(INSTALL_DATA) $(srcdir)/eign $(common_words_file); \
-	fi
+	$(mkinstalldirs) $(DESTDIR)$(datadir)
+	$(mkinstalldirs) $(DESTDIR)$(dataprogramdir)
+	$(mkinstalldirs) $(DESTDIR)$(datasubdir)
+	rm -f $(DESTDIR)$(common_words_file);
+	$(INSTALL_DATA) $(srcdir)/eign $(DESTDIR)$(common_words_file);
 
 uninstall_sub:
 	-rm -f $(common_words_file)
diff -Nru groff-1.19.orig/tmac/Makefile.sub groff-1.19/tmac/Makefile.sub
--- groff-1.19.orig/tmac/Makefile.sub	2003-04-08 19:47:27.000000000 +0200
+++ groff-1.19/tmac/Makefile.sub	2003-04-30 09:44:41.000000000 +0200
@@ -46,41 +46,31 @@
 
 install_data: $(NORMALFILES) $(SPECIALFILES) man.local \
               stamp-strip stamp-wrap stamp-sed
-	-test -d $(tmacdir) || $(mkinstalldirs) $(tmacdir)
-	-test -d $(systemtmacdir) || $(mkinstalldirs) $(systemtmacdir)
-	-test -d $(localtmacdir) || $(mkinstalldirs) $(localtmacdir)
+	$(mkinstalldirs) $(DESTDIR)$(tmacdir)
+	$(mkinstalldirs) $(DESTDIR)$(systemtmacdir)
+	$(mkinstalldirs) $(DESTDIR)$(localtmacdir)
 	if test -n "$(tmac_wrap)"; then \
 	  for m in ""$(tmac_wrap); do \
-	    $(INSTALL_DATA) $$m-wrap $(systemtmacdir)/$$m.tmac; \
+	    $(INSTALL_DATA) $$m-wrap $(DESTDIR)$(systemtmacdir)/$$m.tmac; \
 	  done; \
 	fi
 	for f in $(NORMALFILES); do \
-	  rm -f $(tmacdir)/$$f; \
-	  $(INSTALL_DATA) $(srcdir)/$$f $(tmacdir)/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)$(tmacdir)/$$f; \
 	done
-	-rm -f $(tmacdir)/$(tmac_s_prefix)s.tmac
-	$(INSTALL_DATA) $(srcdir)/s.tmac $(tmacdir)/$(tmac_s_prefix)s.tmac
-	-rm -f $(tmacdir)/ms.tmac
-	$(INSTALL_DATA) ms.tmac-sed $(tmacdir)/ms.tmac
-	-rm -f $(tmacdir)/$(tmac_an_prefix)an.tmac
-	$(INSTALL_DATA) $(srcdir)/an.tmac $(tmacdir)/$(tmac_an_prefix)an.tmac
-	-rm -f $(tmacdir)/man.tmac
-	$(INSTALL_DATA) man.tmac-sed $(tmacdir)/man.tmac
-	-rm -f $(tmacdir)/www.tmac
-	$(INSTALL_DATA) www.tmac-sed $(tmacdir)/www.tmac
+	$(INSTALL_DATA) $(srcdir)/s.tmac $(DESTDIR)$(tmacdir)/$(tmac_s_prefix)s.tmac
+	$(INSTALL_DATA) ms.tmac-sed $(DESTDIR)$(tmacdir)/ms.tmac
+	$(INSTALL_DATA) $(srcdir)/an.tmac $(DESTDIR)$(tmacdir)/$(tmac_an_prefix)an.tmac
+	$(INSTALL_DATA) man.tmac-sed $(DESTDIR)$(tmacdir)/man.tmac
+	$(INSTALL_DATA) www.tmac-sed $(DESTDIR)$(tmacdir)/www.tmac
 	for f in $(STRIPFILES); do \
-	  rm -f $(tmacdir)/$$f; \
-	  $(INSTALL_DATA) $$f-s $(tmacdir)/$$f; \
+	  $(INSTALL_DATA) $$f-s $(DESTDIR)$(tmacdir)/$$f; \
 	done
-	-test -d $(mdocdir) || $(mkinstalldirs) $(mdocdir)
+	$(mkinstalldirs) $(DESTDIR)$(mdocdir)
 	for f in $(MDOCFILES); do \
-	  rm -f $(mdocdir)/$$f; \
-	  $(INSTALL_DATA) $$f-s $(mdocdir)/$$f; \
+	  $(INSTALL_DATA) $$f-s $(DESTDIR)$(mdocdir)/$$f; \
 	done
-	-test -f $(localtmacdir)/man.local || \
-	  $(INSTALL_DATA) $(srcdir)/man.local $(localtmacdir)/man.local
-	-test -f $(localtmacdir)/mdoc.local || \
-	  $(INSTALL_DATA) mdoc.local-s $(localtmacdir)/mdoc.local
+	$(INSTALL_DATA) $(srcdir)/man.local $(DESTDIR)$(localtmacdir)/man.local
+	$(INSTALL_DATA) mdoc.local-s $(DESTDIR)$(localtmacdir)/mdoc.local
 
 stamp-strip: $(STRIPFILES) $(MDOCFILES) mdoc.local
 	for f in $(STRIPFILES) $(MDOCFILES) mdoc.local; do \
--- groff-1.19.1/contrib/grap2graph/Makefile.sub.orig	2003-07-02 17:35:01.000000000 +0200
+++ groff-1.19.1/contrib/grap2graph/Makefile.sub	2004-07-20 22:37:45.706482464 +0200
@@ -11,9 +11,9 @@
 	chmod +x $@
 
 install_data: grap2graph
-	-test -d $(bindir) || $(mkinstalldirs) $(bindir)
-	-rm -f $(bindir)/grap2graph
-	$(INSTALL_SCRIPT) grap2graph $(bindir)/grap2graph
+	-test -d $(DESTDIR)$(bindir) || $(mkinstalldirs) $(DESTDIR)$(bindir)
+	-rm -f $(DESTDIR)$(bindir)/grap2graph
+	$(INSTALL_SCRIPT) grap2graph $(DESTDIR)$(bindir)/grap2graph
 
 uninstall_sub:
 	-rm -f $(bindir)/grap2graph
