diff -Nru groff-1.19.orig/doc/groff.texinfo groff-1.19/doc/groff.texinfo
--- groff-1.19.orig/doc/groff.texinfo	2003-04-29 09:20:21.000000000 +0200
+++ groff-1.19/doc/groff.texinfo	2003-04-30 10:01:16.000000000 +0200
@@ -8,7 +8,7 @@
 @c
 
 @c %**start of header (This is for running Texinfo on a region.)
-@setfilename groff
+@setfilename groff.info
 @settitle The GNU Troff Manual
 @setchapternewpage odd
 @footnotestyle separate
@@ -361,9 +361,9 @@
 @c Note: We say `Roman numerals' but `roman font'.
 
 
-@dircategory Typesetting
+@dircategory Text Processing Tools:
 @direntry
-* Groff: (groff).               The GNU troff document formatting system.
+* Groff: (groff).			The GNU troff document formatting system.
 @end direntry
 
 
